lib/flags.ml

```diff
@@ -11,17 +11,14 @@
 (* If [restore] is false, whenever [f] modifies the ref, we will
    preserve the modification. *)
 let with_modified_ref ?(restore=true) r nf f x =
-  let old_ref = !r in r := nf !r;
-  try
-    let pre = !r in
-    let res = f x in
+  let acquire x = let old_ref = !r in r := nf !r; (old_ref, !r, x) in
+  let release (old_ref, pre, _) =
     (* If r was modified don't restore its old value *)
     if restore || pre == !r then r := old_ref;
```

Comment:
```suggestion
```
While randomly looking at this diff, I noticed that this looks useless since the `r := old_ref` below seems to always run.

Reply:
Good point, I need to update this PR to new internal version I'm using on my stress-testing, but indeed this looks fishy and deserves a closer look, thanks for the hint!

Reply:
The `:= old_ref` below is only when `f` produces an exception

but IDK if the restore:false case is useful indeed